scipy/stats/tests/test_multivariate.py

```diff
@@ -1964,6 +1965,40 @@ def test_to_corr(self):
         assert_allclose(m[0,0], 1)
 
 
+class TestRandomDirection:
+    @pytest.mark.parametrize("dim", [1, 2, 3, 4])
+    @pytest.mark.parametrize("size", [5, (5, 4)])
+    def test_samples(self, dim, size):
+        # test that samples have correct shape and norm 1
+        random_direction_dist = random_direction(dim)
+        samples = random_direction_dist.rvs(size)
+        if isinstance(size, int):
+            size = (size, )
+        expected_shape = size + (dim, )
+        assert samples.shape == expected_shape
+        norms = np.linalg.norm(samples, axis=-1)
+        expected_norms = np.ones(size)
+        assert_allclose(norms, expected_norms)
+
+    def test_invalid_dim(self):
+        assert_raises(ValueError, random_direction.rvs, None)
+        assert_raises(ValueError, random_direction.rvs, 0)
+        assert_raises(ValueError, random_direction.rvs, (2, 2))
+        assert_raises(ValueError, random_direction.rvs, 2.5)
+
+    def test_frozen_distribution(self):
+        dim = 5
+        frozen = random_direction(dim)
+        frozen_seed = random_direction(dim, seed=514)
```

Comment:
There are other maintainers that would want `seed` to be a Generator. I'll leave it up to you.